cpp/src/io/utilities/parsing_utils.cuh

```diff
@@ -286,6 +286,90 @@ __inline__ __device__ T parse_numeric(const char* begin,
   return value * sign;
 }
 
+/**
+ * @brief Lexicographically compare digits in input against string
+ * representing an integer
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @param golden The pointer to beginning of character string representing
+ * the value to be compared against
+ * @return bool True if integer represented by character string is less
+ * than or equal to golden data
+ */
+template <int N>
+__device__ __inline__ bool less_equal_than(const char* data, const char (&golden)[N])
+{
+  auto mismatch_pair = thrust::mismatch(thrust::seq, data, data + N - 1, golden);
+  if (mismatch_pair.first != data + N - 1) {
+    return *mismatch_pair.first <= *mismatch_pair.second;
+  } else {
+    // Exact match
+    return true;
+  }
+}
+
+/**
+ * @brief Determine which counter to increment when a sequence of digits
+ * and a parity sign is encountered.
+ *
+ * @param raw_data The pointer to beginning of character string
+ * @param digit_count Total number of digits
+ * @param stats Reference to structure with counters
+ * @return Pointer to appropriate counter that belong to
+ * the interpreted data type
+ */
+__device__ __inline__ cudf::size_type* infer_integral_field_counter(char const* data_begin,
+                                                                    char const* data_end,
+                                                                    bool is_negative,
+                                                                    column_type_histogram& stats)
+{
+  static constexpr char uint64_max_abs[] = "18446744073709551615";
+  static constexpr char int64_min_abs[]  = "9223372036854775808";
+  static constexpr char int64_max_abs[]  = "9223372036854775807";
+
+  int32_t digit_count = data_end - data_begin;
```

Comment:
very edge-case-y, should we use `ptrdiff_t` in case of very long strings?
```suggestion
  ptrdiff_t digit_count = data_end - data_begin;
```

Reply:
Yes, `ptrdiff_t` is the better option here. But I think `auto` is the best option, no need to think about what the type of pointer diff is.